src/coreclr/runtime-prereqs.proj

```diff
@@ -6,7 +6,7 @@
     <RuntimeVersionFile>$(ArtifactsObjDir)runtime_version.h</RuntimeVersionFile>
     <NativeSourceLinkFile>$(ArtifactsObjDir)native.sourcelink.json</NativeSourceLinkFile>
     <VerifySourceLinkFileExists>false</VerifySourceLinkFileExists>
-    <VerifySourceLinkFileExists Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(DotNetBuildFromSource)' != 'true'">true</VerifySourceLinkFileExists>
+    <VerifySourceLinkFileExists Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</VerifySourceLinkFileExists>
```

Comment:
Just for mitigating risk, I'd remove changes here. It only applies to windows (native sourcelink).

Reply:
This is applicable to Linux as well - `NativeSourceLinkFile` points to a file that exists in Linux build, so we should check for file existence.

Reply:
It will never use it though. It's only passed to MSVC here https://github.com/dotnet/runtime/blob/03f16f1a85fca5fe659338b284d70571990ee935/eng/native/configurecompiler.cmake#L81

Reply:
> It will never use it though. It's only passed to MSVC here
> 
> https://github.com/dotnet/runtime/blob/03f16f1a85fca5fe659338b284d70571990ee935/eng/native/configurecompiler.cmake#L81

Interesting. I might be missing when the file gets generated, but it did get created on Linux and had the following content. Should the file even be created on Linux?

```
{"documents":{"/src/source.online/src/runtime/artifacts/source-build/self/src/*":"https://raw.githubusercontent.com/nikolamilosavljevic/runtime/056cf9bd4c52f7bb8fff74bc47d9c6d2562c3e9c/*"}}
```

Reply:
yeah, the file gets created always. It's used by native only on Windows. It gets used for managed projects mostly.

Reply:
Thanks! Since this is useful for Windows only, it makes sense to remove it on source-build and revert the change I proposed.

Reply:
Based on this discussion wouldn't it be better to change this to be a windows specific check?  e.g. only enabled for windows CI?  The point is this really shouldn't be a source-build specific check.  I understand however if the desire is to limit changes this late in the release cycle.